File: model/src/main/kotlin/net/dungeonhub/model/PurgeTypeRoleModel.java
package net.dungeonhub.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import me.taubsie.dungeonhub.common.entity.model.Model;
import me.taubsie.dungeonhub.common.model.discord_role.DiscordRoleModel;
import me.taubsie.dungeonhub.common.model.purge_type.PurgeTypeModel;

@Getter
@AllArgsConstructor
public class PurgeTypeRoleModel implements Model {
    private PurgeTypeModel purgeTypeModel;
    private DiscordRoleModel discordRoleModel;
}

File: model/src/main/kotlin/net/dungeonhub/model/purge_type/PurgeTypeRoleModel.kt
package net.dungeonhub.model.purge_type

import net.dungeonhub.structure.model.Model
import net.dungeonhub.model.discord_role.DiscordRoleModel

class PurgeTypeRoleModel(
    val purgeTypeModel: SimplePurgeTypeModel,
    val discordRoleModel: DiscordRoleModel
) : Model